secure include for .NET AKV (#923)
dotnet/AspNetDocs: aspnet/includes/connection_string.md

---
author: tdykstra
ms.author: tdykstra
ms.date: 10/16/2024
ms.topic: include
---
<!-- ms.sfi.ropc: t -->
> [!WARNING]

> > This article shows the use of connection strings. With a local database the user doesn't have to be authenticated, but in production, connection strings sometimes include a password to authenticate. A resource owner password credential (ROPC) is a security risk that should be avoided in production databases. Production apps should use the most secure authentication flow available.

Reviewer: Thanks for calling this out. I think this is as gentle a way of letting the customer know that connection strings are sub-optimal prior to showing them how to deal with the situation in which they can't be avoided.

> > This article shows the use of connection strings. With a local database the user doesn't have to be authenticated, but in production, connection strings sometimes include a password to authenticate. A resource owner password credential (ROPC) is a security risk that should be avoided in production databases. Production apps should use the most secure authentication flow available.

Reviewer 2: @tdykstra ... There are a couple of structural problems with the first sentence (missing comma, missing semicolon). In addition, here's a version that avoids some state verbs ...

```suggestion
> This article shows the use of connection strings. Using a local database, user authentication isn't required; but in production, connection strings might include a password to authenticate. A resource owner password credential (ROPC) is a security risk to avoid in production databases. Production apps should use the most secure authentication flow available.
```
